client/app/telecom/pack/xdsl/modem/service/pack-xdsl-modem-service.controller.js

```diff
@@ -0,0 +1,359 @@
+    return this.OvhApiXdsl.Modem().ContentSharing().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      contentSharing: status,
+    }).$promise.then((data) => {
+      this.mediator.setTask('changeModemContentSharingStatus');
+      this.service.contentSharingTmp = this.service.contentSharing;
+      if (this.service.isFtp) {
+        this.service.ftp = !this.service.ftp;
+        this.updateFTP(status);
+      }
+      if (this.service.isUpnp) {
+        this.service.upnp = !this.service.upnp;
+        this.updateUPnP(status);
+      }
+      this.TucToast.success(this.$translate.instant('xdsl_modem_service_content_sharing_doing'));
+      return data;
+    }).catch((err) => {
+      this.service.contentSharing = this.service.contentSharingTmp;
+      this.TucToast.error(this.$translate.instant('xdsl_modem_service_content_sharing_an_error_ocurred'));
+      return this.$q.reject(err);
+    });
+  }
+
+  getFTP() {
+    return this.OvhApiXdsl.Modem().Ftp().v6().get({
+      xdslId: this.$stateParams.serviceName,
+    }).$promise.then(({ data }) => {
+      if (data === 'enabled') {
+        this.service.ftp = true;
+      } else {
+        this.service.ftp = false;
+      }
+      this.service.ftTmp = this.service.ftp;
+      this.service.isFtp = true;
+      return data;
+    }).catch(() => {
+      this.service.isFtp = false;
+    });
+  }
+
+  changeFTP(value) {
+    if (value) {
+      this.updateFTP('enabled');
+    } else {
+      this.updateFTP('disabled');
+    }
+  }
+
+  updateFTP(status) {
+    return this.OvhApiXdsl.Modem().Ftp().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      ftp: status,
+    }).$promise.then((data) => {
+      this.mediator.setTask('changeModemFtpStatus');
+      this.service.ftTmp = this.service.ftp;
+      this.TucToast.success(this.$translate.instant('xdsl_modem_service_ftp_doing'));
+      return data;
+    }).catch((err) => {
+      this.service.ftp = this.service.ftTmp;
+      this.TucToast.error(this.$translate.instant('xdsl_modem_service_ftp_an_error_ocurred'));
+      return this.$q.reject(err);
+    });
+  }
+
+  getIpsecAlg() {
+    return this.OvhApiXdsl.Modem().IpsecAlg().v6().get({
+      xdslId: this.$stateParams.serviceName,
+    }).$promise.then(({ data }) => {
+      if (data === 'enabled') {
+        this.service.ipsecAlg = true;
+      } else {
+        this.service.ipsecAlg = false;
+      }
+      this.service.ipsecAlgTmp = this.service.ipsecAlg;
+      this.service.isIpsecAlg = true;
+      return data;
+    }).catch(() => {
+      this.service.isIpsecAlg = false;
+    });
+  }
+
+  changeIpsecAlg(value) {
+    if (value) {
+      this.updateIpsecAlg('enabled');
+    } else {
+      this.updateIpsecAlg('disabled');
+    }
+  }
+
+  updateIpsecAlg(status) {
+    return this.OvhApiXdsl.Modem().IpsecAlg().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      ipsecAlg: status,
+    }).$promise.then((data) => {
```

Comment:
According to the API response class:

```suggestion
    }).$promise.then((xdslTask) => {
```